src/lib.rs

```diff
@@ -1093,51 +1089,15 @@ impl<T: Activated + ?Sized> Capture<T> {
         }
     }
 
-    #[cfg(feature = "capture-stream")]
-    fn next_noblock<'a>(
-        &'a mut self,
-        cx: &mut core::task::Context,
-        fd: &mut tokio::io::unix::AsyncFd<stream::SelectableFd>,
-    ) -> Result<Packet<'a>, Error> {
-        let ready = fd.poll_read_ready(cx);
-        if ready.is_pending() {
-            Err(IoError(io::ErrorKind::WouldBlock))
-        } else {
-            match self.next() {
-                Ok(p) => Ok(p),
-                Err(TimeoutExpired) => {
-                    // Per https://docs.rs/tokio/1.12.0/tokio/io/unix/struct.AsyncFd.html
-                    // ... it is critical to ensure that this ready flag
-                    // is cleared when (and only when) the file descriptor
-                    // ceases to be ready.
-                    //
-                    if let Ready(Ok(mut guard)) = ready {
-                        guard.clear_ready();
-                        #[allow(unused_must_use)]
-                        {
-                            fd.poll_read_ready(cx);
-                        }
-                    }
-
-                    Err(IoError(io::ErrorKind::WouldBlock))
-                }
-                Err(e) => Err(e),
-            }
-        }
-    }
-
-    #[cfg(feature = "capture-stream")]
+    #[cfg(all(unix, feature = "capture-stream"))]
```

Comment:
I somehow opposed to that, if a user require a feature disable it if not on unix will lead to uncepted error message such as "not find" instead of "can't compile on windows". This is misleading.

Reply:
Yup, I will revert this change. (Would be nice to be able to specify in the cargo manifest that some feature has some cfg requirement.)